File: sdks/php/src/Subscriptions/Requests/ChangePlanDto.php
<?php

namespace NovaBilling\Subscriptions\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;

class ChangePlanDto extends JsonSerializableType
{
    /**
     * @var string $newPlanId New plan ID
     */
    #[JsonProperty('newPlanId')]
    public string $newPlanId;

    /**
     * @var ?bool $prorate Whether to prorate charges
     */
    #[JsonProperty('prorate')]
    public ?bool $prorate;

    /**
     * @param array{
     *   newPlanId: string,
     *   prorate?: ?bool,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->newPlanId = $values['newPlanId'];
        $this->prorate = $values['prorate'] ?? null;
    }
}

File: sdks/csharp/src/NovaBilling/Subscriptions/Requests/ChangePlanDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ChangePlanDto
{
    /// <summary>
    /// Subscription ID
    /// </summary>
    [JsonIgnore]
    public required string Id { get; set; }

    /// <summary>
    /// New plan ID
    /// </summary>
    [JsonPropertyName("newPlanId")]
    public required string NewPlanId { get; set; }

    /// <summary>
    /// Whether to prorate charges
    /// </summary>
    [JsonPropertyName("prorate")]
    public bool? Prorate { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
